Tools/paletteable sprite editor/js/editor.js

```diff
@@ -0,0 +1,473 @@
+
+        $(".swatch").each(function(index, swatchDiv) {
+            if (index >= deletedIndex) {
+                const nextSwatch = $(swatchDiv).next();
+                if (nextSwatch.length > 0 ) {
+                    const nextColor = nextSwatch.children("input[type='color']").val();
+                    const newDiv = createSwatchDiv(index);
+
+                    newDiv.children("input[type='color']").val(nextColor);
+                    newDiv.insertBefore(swatchDiv);
+                }
+                swatchDiv.remove();
+            }
+        });
+
+        var wantedAIndex = Math.min(globalPaletteSize-1,aSelectedIndex);
+        var wantedBIndex = Math.min(globalPaletteSize-1,bSelectedIndex);
+        if (wantedAIndex > deletedIndex) {
+            wantedAIndex--;
+        }
+        if (bSelectedIndex > deletedIndex) {
+            wantedBIndex--;
+        }
+
+        console.log(wantedAIndex);
+        console.log(wantedBIndex);
+        console.log($(".swatch>.a-radio").eq(wantedAIndex));
+        $(".swatch>.a-radio").eq(wantedAIndex).prop('checked', true);
+        $(".swatch>.b-radio").eq(wantedBIndex).prop('checked', true);
+
+        swatches.splice(deletedIndex,1);
+
+        for (var p=drawPosRaw.length-1; p>=0; p--)
+        {
+            const logicalBefore = drawPosRaw[p].rawColorObject.toLogical(oldGlobalPaletteSize);
+            var indexA = logicalBefore.indexA;
+            var indexB = logicalBefore.indexB;
+
+            if (indexA == deletedIndex || indexB == deletedIndex) {
+                drawPosRaw.splice(p,1);
+            } else {
+                indexA = Math.min(globalPaletteSize-1, indexA);
+                indexB = Math.min(globalPaletteSize-1, indexB);
+                if (indexA > deletedIndex) { indexA--;}
+                if (indexB > deletedIndex) { indexB--;}
+                const interp = logicalBefore.interp;
+                const newColor = logicalToColor(indexA, indexB, interp, globalPaletteSize);
+                drawPosRaw[p].rawColorObject = newColor;
+                drawPosRaw[p].rawColor = newColor.asHex();
+            }
+        }
+
+        onPaletteChange();
+    }
+
+    window.requestAnimationFrame(render);
+    $("#swatch-collection").on("change", "input[type='radio'],input[type='color']", onPaletteChange);
+    $("#swatch-collection").on("click", ".delete-swatch", deleteSwatch);
+
+    document.getElementById("mixRange").addEventListener("input", onPaletteChange);
+
+    renderCanvas.addEventListener("mousemove", recordMouseMovement);
+    renderCanvas.addEventListener("mousedown", startDrawing);
+    renderCanvas.addEventListener("mouseup", stopDrawing);
+    renderCanvas.addEventListener("contextmenu", clearPixel);
+    document.getElementById("loadRaw").addEventListener("click", loadImage);
+
+    function recordMouseMovement(event) {
+        mouse = getMousePos(event);
+    }
+
+    function startDrawing(event) {
+        if (event.button == 0) {
```

Comment:
Codacy found an issue: [Use ===/!== to compare with true/false or Numbers](https://app.codacy.com/gh/unitystation/unitystation/pullRequest?prid=6395467)